src/Classes/TradeQuery.lua

```diff
@@ -0,0 +1,897 @@
+	if activeSlotRef and activeSlotRef.pbURL ~= "" and activeSlotRef.pbURL ~= nil then
+		controls['uri'..str_cnt]:SetText(activeSlotRef.pbURL)
+	else
+		controls['uri'..str_cnt]:SetText("<PASTE TRADE URL FOR>: " .. slotTbl.name)
+	end
+	controls['uri'..str_cnt].tooltipFunc = function(tooltip)
+		tooltip:Clear()
+		if controls['uri'..str_cnt].buf:find('^https://www.pathofexile.com/trade/search/') ~= nil then
+			tooltip:AddLine(16, "CTRL click to open in web-browser or click 'Price Item' to do it in PoB")
+			tooltip:AddLine(16, "")
+			tooltip:AddLine(14, colorCodes.NEGATIVE .. "NOTE: you will need to re-sort until GGG fixes")
+		end
+	end
+	controls['priceButton'..str_cnt] = new("ButtonControl", {"TOPLEFT",controls['uri'..str_cnt],"TOPLEFT"}, 500 + 8, 0, 100, row_height, "Price Item", function()
+		self:PublicTrade(controls['uri'..str_cnt].buf, slotTbl, controls, str_cnt)
+	end)
+	controls['priceButton'..str_cnt].enabled = function()
+		local validURL = controls['uri'..str_cnt].buf:find('^https://www.pathofexile.com/trade/search/') ~= nil
+		if not activeSlotRef and slotTbl.ref then
+			self.itemsTab.activeItemSet[slotTbl.ref] = { pbURL = "" }
+			activeSlotRef = self.itemsTab.activeItemSet[slotTbl.ref]
+		end
+		if validURL then
+			activeSlotRef.pbURL = controls['uri'..str_cnt].buf
+		elseif controls['uri'..str_cnt].buf == "" then
+			activeSlotRef.pbURL = ""
+		end
+		return validURL and self:PriceBuilderCanSearch(controls) and self:PriceBuilderCanFetch(controls)
+	end
+	controls['resultIndex'..str_cnt] = new("EditControl", {"TOPLEFT",controls['priceButton'..str_cnt],"TOPLEFT"}, 100 + 8, 0, 60, row_height, "#", nil, "%D", 3, function(buf)
+		controls['resultIndex'..str_cnt].buf = tostring(m_min(m_max(tonumber(buf) or 1, 1), self.sortedResultTbl[str_cnt] and #self.sortedResultTbl[str_cnt] or 1))
+	end)
+	controls['resultIndex'..str_cnt].tooltipFunc = function(tooltip)
+		if tooltip:CheckForUpdate(controls['resultIndex'..str_cnt].buf) then
+			self:SetFetchResultReturn(controls, str_cnt, tonumber(controls['resultIndex'..str_cnt].buf))
+		end
+	end
+	controls['resultCount'..str_cnt] = new("EditControl", {"TOPLEFT",controls['resultIndex'..str_cnt],"TOPLEFT"}, 60 + 8, 0, 82, row_height, "^8No Results", nil, nil)
+	controls['priceAmount'..str_cnt] = new("EditControl", {"TOPLEFT",controls['resultCount'..str_cnt],"TOPLEFT"}, 82 + 8, 0, 120, row_height, "Price", nil, nil)
+	controls['priceAmount'..str_cnt].enabled = function()
+		local boolean = #controls['priceAmount'..str_cnt].buf > 0
+		return boolean
+	end
+	controls['importButton'..str_cnt] = new("ButtonControl", {"TOPLEFT",controls['priceAmount'..str_cnt],"TOPLEFT"}, 120 + 8, 0, 100, row_height, "Import Item", function()
+		self.itemsTab:CreateDisplayItemFromRaw(controls['importButtonText'..str_cnt].buf)
+		local item = self.itemsTab.displayItem
+		-- pass "true" to not auto equip it as we will have our own logic
```

Comment:
Because there are three main differences in how it's implemented:
1) I don't want to create an extra click in to "save" the item
2) I don't want to leave the item in the preview-pane requiring an "Add To Build" click
3) I want the item to auto-populate in the right slot when the button is hit (even when an Item is already in that slot)
3a) I differentiate between Left and Right Ring Slots (for example) as well. The default doesn't.

Reply:
Yeah, that makes sense.  I guess I didn't realize the autoEquip doesn't replace an existing item.  I'm not sure if the two undo states are desirable (one that imports the item, one that equips it).  These lines would turn that into just one undo state:
```lua
self.itemsTab:AddItem(item, true)
self.itemsTab:SetDisplayItem()
```